src/projection_client.rs

```diff
@@ -0,0 +1,629 @@
+        options: projections::create_req::Options,
+    ) -> crate::Result<()> {
+        let req = projections::CreateReq {
+            options: Some(options),
+        };
+
+        let mut req = tonic::Request::new(req);
+
+        crate::commands::configure_auth_req(&mut req, credentials.cloned());
+
+        self.client
+            .execute(|handle| async move {
+                let mut client =
+                    projections::projections_client::ProjectionsClient::new(handle.channel);
+                let _ = client.create(req).await?;
+
+                Ok(())
+            })
+            .await
+    }
+
+    pub async fn update_projection<Name>(
+        &self,
+        name: Name,
+        query: String,
+        emit_option: Option<bool>,
+        credentials: Option<&Credentials>,
+    ) -> crate::Result<()>
+    where
+        Name: AsRef<str>,
+    {
+        let options = projections::update_req::Options {
+            name: name.as_ref().to_string(),
+            emit_option: emit_option
+                .map(projections::update_req::options::EmitOption::EmitEnabled)
+                .or(Some(
+                    projections::update_req::options::EmitOption::NoEmitOptions(Empty {}),
+                )),
+            query,
+        };
+
+        let req = projections::UpdateReq {
+            options: Some(options),
+        };
+
+        let mut req = tonic::Request::new(req);
+
+        crate::commands::configure_auth_req(&mut req, credentials.cloned());
+
+        self.client
+            .execute(|handle| async move {
+                let mut client =
+                    projections::projections_client::ProjectionsClient::new(handle.channel.clone());
+
+                let _ = client.update(req).await?;
+
+                Ok(())
+            })
+            .await
+    }
+
+    pub async fn delete_projection<Name>(
+        &self,
+        name: Name,
+        delete_emitted_streams: bool,
+        delete_state_stream: bool,
+        delete_checkpoint_stream: bool,
+        credentials: Option<&Credentials>,
+    ) -> crate::Result<()>
+    where
+        Name: AsRef<str>,
+    {
+        let options = projections::delete_req::Options {
+            name: name.as_ref().to_string(),
+            delete_emitted_streams,
+            delete_state_stream,
+            delete_checkpoint_stream,
+        };
+
+        let req = projections::DeleteReq {
+            options: Some(options),
+        };
+
+        let mut req = tonic::Request::new(req);
+
+        crate::commands::configure_auth_req(&mut req, credentials.cloned());
+
+        self.client
+            .execute(|handle| async move {
+                let mut client =
+                    projections::projections_client::ProjectionsClient::new(handle.channel.clone());
+
+                let _ = client.delete(req).await?;
+
+                Ok(())
+            })
+            .await
+    }
+
+    pub async fn statistics(
+        &self,
+        stats_for: StatsFor,
```

Comment:
**CONCERN:** I'm not sure how close we should be in the statistics API implementation, but .NET gRPC client has separate methods: `ListOneTimeAsync`, `ListContinuousAsync` and `GetStatusAsync`. The NodeJS is similar to Rust, but the API and name are much different `getProjectionStatistics` that allows passing just a projection name.

Reply:
I believe .NET picked that route because of the lack of proper enum support. I agree on having a common experience but would like to keep respective programming language strength too.

Reply:
No longer publicly exposed